evm/src/main/java/org/hyperledger/besu/evm/processor/AbstractMessageProcessor.java

```diff
@@ -146,7 +152,10 @@ protected void revert(final MessageFrame frame) {
    *
    * @param frame The message frame
    */
-  private void completedSuccess(final MessageFrame frame) {
+  private void completedSuccess(final MessageFrame frame, final OperationTracer operationTracer) {
+    if (operationTracer != null) {
```

Comment:
I'd rather see these in the `process` method and these signatures not altered (like was done with the `CODE_SUSPENDED` case).  The `contextEnter` call also occurs there.